pkg/server/admin.go

```diff
@@ -1397,6 +1397,43 @@ func (s *adminServer) checkReadinessForHealthCheck() error {
 		return status.Errorf(codes.Unavailable, "node is shutting down")
 	}
 
+	if !s.server.sqlServer.acceptingClients.Get() {
+		return status.Errorf(codes.Unavailable, "node is not accepting SQL clients")
+	}
+
+	// Also check the availability of the SQL service.
```

Comment:
An alternative design to consider:

1. Here we do a basic minimum check: `!s.server.sqlServer.acceptingClients.Get`. If the probe fails, we halt the upgrade.
2. In the CC / k8s operator upgrade automation, we check **time-series metrics** during the upgrade. If the metrics look bad, e.g. if they look different between the updated node & a node running the old version, we halt the upgrade.

Why check metrics? Well, the aggregations we can do in prometheus may lead to higher signal to noise than making a local decision on the node itself. Also, maybe it's more extensible? When we get access to new & rich time-series data, it's easy to integrate into the upgrade process.

Lastly, it side steps the worries in my above comment, as 2 won't affect k8s load balancing decisions by design.